### ISSUE TITLE
custom_components.hacs.config_flow::HacsFlowHandler calls async_show_progress without passing a progress task

### ISSUE THREAD

**Issue by jmartens** [hacs/integration#3649]
## System Information

version | core-2024.5.0.dev0
-- | --
installation_type | Unsupported Third Party Container
dev | true
hassio | false
docker | true
user | vscode
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 5.15.146.1-microsoft-standard-WSL2
arch | x86_64
timezone | Europe/Amsterdam
config_dir | /workspaces/core/config

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 0
views | 0
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | April 16, 2024 at 11:38 AM
-- | --
current_recorder_run | April 16, 2024 at 1:54 PM
estimated_db_size | 0.22 MiB
database_engine | sqlite
database_version | 3.40.1

</details>

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [ ] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the backend (integration part) of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [open](https://github.com/hacs/integration/issues) or issues [pending release](https://github.com/hacs/integration/issues?q=is%3Aissue+is%3Aclosed+sort%3Aupdated-desc+milestone%3Anext).

### Describe the issue

This appears in the HACS logs when installing it:

```
2024-04-16 11:57:15.006 WARNING (MainThread) [homeassistant.data_entry_flow] custom_components.hacs.config_flow::HacsFlowHandler calls async_show_progress without passing a progress task, this is not valid and will break in Home Assistant Core 2024.8. Please report it to the custom integration author
```

### Reproduction steps

1. Spin up home assistant dev container
2. Install HACS using default installation instruction using wget and script
3. Add HACS integration
4. Observe logs for warning
...

### Debug logs

```text
N/A
```

### Diagnostics dump

N/A

**Comment by jmartens** [comment]
For details see https://developers.home-assistant.io/blog/2024/01/11/async-show-progress-changes/

**Comment by seryte** [comment]
+1

**Comment by jmartens** [comment]
> +1

Please don't do this but do as asked:

> If you have an issue identical to this, do **not** add comments like "same here", "i have this too", instead add a 👍 reaction to the issue description. Thanks! 👍

**Comment by dhover** [comment]
I have added HACS to a clean HA 2024.5.0b0 but the install got stuck on the dialog "Waiting for device activation". On the github side everything seems to be ok. In the logfile I see also the message mentioned in this issue.

**Comment by Goran58** [comment]
> I have added HACS to a clean HA 2024.5.0b0 but the install got stuck on the dialog "Waiting for device activation". On the github side everything seems to be ok. In the logfile I see also the message mentioned in this issue.

Had the same issue. After fixing the async_show-progress warning, activation worked for me. Until an official fix will be released you can locally add a line to HACS's config_flow.py:128 accoording to this:

```
return self.async_show_progress(
            step_id="device",
            progress_action="wait_for_device",
            **progress_task=self.activation_task,**
            description_placeholders={
                "url": OAUTH_USER_LOGIN,
                "code": self._registration.user_code,
            },
        )
```

**Comment by jmartens** [comment]
> Had the same issue. After fixing the async_show-progress warning, activation worked for me. Until an official fix will be released you can locally add a line (progress_task=...) to HACS's config_flow.py:128 accoording to this:

Did you raise a ticker or PR for this? If so do you have a reference to it?

**Comment by Goran58** [comment]
Pull request #3582

**Comment by hongkongkiwi** [comment]
I could not activate at all following all instructions, this fixed it for me too straight away. I think it would be important to integrate this fix.

**Comment by laojr1983** [comment]
It's necessary you restart o HA.

**Comment by blorby** [comment]
> progress_task=self.activation_task,

thank you! that was a huge help!
the file location is: /homeassistant/cusom_components/hacs/config_flow.py

**Comment by mroffbeat** [comment]
Geez.. Thank you SO much, drove me wild.. Restoring, removing, redownloading.. trying, google'n.. Very frustrating

**Comment by AdamsGH** [comment]
[Goran58](https://github.com/Goran58), you are life saver. Holy fck, 5 hours of life...

**Comment by xandalo** [comment]
Thank you so much. I was on this for 2 days now.

**Comment by Servantis** [comment]
Thank you.

**Comment by Rexyocouch** [comment]
Adding the above fix from [Goran58](https://github.com/Goran58) fixed it for me as well, I'm running on a fresh install Raspberry 5.

**Comment by chvima** [comment]
I can confirm that after the hotfix by @Goran58 on the file:
`/homeassistant/custom_components/hacs/config_flow.py` 
works

**Comment by GyHUN95** [comment]
I can confirm Goran's sokutions too, it works on 2024.5 HA lxc on proxmox 8.2! 👍

**Comment by abliss** [comment]
I also hit this. It's extra frustrating because the docs at https://hacs.xyz/docs/configuration/basic tell me to press a "submit" button which does not exist .

**Comment by drucknase** [comment]
thanks a lot for the workaround. I found it today after 2 days of search for the problem!! 
Core 2024.5.1, Supervisor 2024.04.4, Operating System 12.2, Frontend 20240501.0

**Comment by Zero-Circle** [comment]
This drove me insane for a couple of hours tonight, 
Thanks @Goran58, had to learn how to use nano and after a couple of attempts I got it working!
Core 2024.5.1 | Supervisor 2024.04.4 | Operating System 12.2 | Frontend 20240501.0

**Comment by jmartens** [comment]
@ludeeus just merged #3582. I think this can be closed, right @ludeeus?
